### PR TITLE
Add Theme module

### PR DESCRIPTION
**Pull request by driftingly**
This PR adds the Theme module.

> [!NOTE]  
> Themes will continue to be refined after I get access to the staging CMS and can verify current fields and functionality. 

The database seeder script is used to pull in content from these available JSON files:

- http://journeymaker.artic.edu/sites/default/files/json/data-en.json
- http://journeymaker.artic.edu/sites/default/files/json/data-es.json
- http://journeymaker.artic.edu/sites/default/files/json/data-zh-hans.json

Update `.env` file with the following sources:

```bash
LOCALE_DATA_EN="http://journeymaker.artic.edu/sites/default/files/json/data-en.json"
LOCALE_DATA_ES="http://journeymaker.artic.edu/sites/default/files/json/data-es.json"
LOCALE_DATA_ZH="http://journeymaker.artic.edu/sites/default/files/json/data-zh-hans.json"
```

You can run the seeder using:

```bash
php artisan migrate:fresh --seed
```

The seeder creates the following admin account:
- `admin@artic.edu`/`password`

> [!IMPORTANT]
> **Known Issues**:
> When creating a new Theme, the form input for `title` allows a maximum of 23 characters.  I was unable to set this same limitation when updating the Theme. If a title is updated to more than 23 characters, there is a validation error, but it's not visible until you click the title to edit it. 

<details>

<summary>Screenshots</summary>

Screenshot 2024-04-03 at 4 32 24 PM

--- 

Screenshot 2024-04-03 at 4 39 17 PM

--- 

Screenshot 2024-04-03 at 4 51 30 PM

--- 

Screenshot 2024-04-03 at 4 51 41 PM

</details>

### CODE REVIEW

File: app/Models/Theme.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    public $mediasParams = [
+        'icon' => [
+            'default' => [
+                [
+                    'name' => 'default',
+                    'ratio' => 1,
+                    'minValues' => [
+                        'width' => 76,
+                        'height' => 76,
```
Comment (zachgarwood) [inline app/Models/Theme.php]: In the `ThemeRequest` validation rules, the min height/width is 75.

File: app/Models/Theme.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    'name' => 'default',
+                    'ratio' => 0,
+                    'minValues' => [
+                        'width' => 1125,
+                        'height' => 1500,
+                    ],
+                ],
+            ],
+        ],
+        'bgs' => [
```
Comment (zachgarwood) [inline app/Models/Theme.php]: This seems like an unnecessary abbreviation. Please make this `background` instead.

**Review by zachgarwood — CHANGES_REQUESTED** [review]
Only two minor change request

File: database/seeders/DatabaseSeeder.php [hunk 3]
```diff
@@ -2,21 +2,136 @@
...
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Str;
 
 class DatabaseSeeder extends Seeder
 {
+    private array $localeSources = [
+        'en' => 'http://journeymaker.artic.edu/sites/default/files/json/data-en.json',
+        'es' => 'http://journeymaker.artic.edu/sites/default/files/json/data-es.json',
+        'zh' => 'http://journeymaker.artic.edu/sites/default/files/json/data-zh-hans.json',
```
Comment (nikhiltri) [inline database/seeders/DatabaseSeeder.php]: As a rule we generally don't commit domain names to code. Move these URLs to `.env` and retrieve them through `config()`.

**Review by nikhiltri — CHANGES_REQUESTED** [review]

File: app/Models/Theme.php (hunk 2 above)
Comment (driftingly) [reply app/Models/Theme.php]: Good call. I had initially used `bgs` to mirror the output JSON.

File: app/Models/Theme.php (hunk 1 above)
Comment (driftingly) [reply app/Models/Theme.php]: Good catch. I've updated these to be the same as the validation rules and made a note to address this duplication.

File: database/seeders/DatabaseSeeder.php (hunk 3 above)
Comment (driftingly) [reply database/seeders/DatabaseSeeder.php]: Added a `journeymaker` config file and update seeder to pull from there.

**Review by nikhiltri — APPROVED** [review]
Looks good! ⭐️

**Review by zachgarwood — APPROVED** [review]

**Pull request merged**